src/AutoRest.CSharp/Common/Generation/Writers/SerializationWriter.cs

```diff
@@ -201,7 +201,7 @@ private void WriteJsonDeserialize(CodeWriter writer, SchemaObjectType model, Jso
                     }
                 }
 
-                if (model.Declaration.IsAbstract)
+                if (model.Declaration.IsAbstract || model.IsAbstract)
                 {
                     writer.Line($"throw new {typeof(NotSupportedException)}(\"Deserialization of abstract type '{model.Type}' not supported.\");");
```

Comment:
How about:
Abstract class '{model.Type}' cannot be instantiated.

Reply:
Data Plane also uses this message, so I prefer to leave it as it is.